### PR TITLE
[ENG-7222][docs] add resource class related documentation for EAS Build

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

The resource class documentation wasn't properly updated in a while.

# How

Update information about resource classes used in EAS Build.

# Test Plan

Test locally

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/build-reference/infrastructure.mdx [hunk 1]
```diff
@@ -29,9 +29,11 @@ When selecting an image for the build you can use the full name provided below o
 
 ## Android build server configurations
 
-Android workers run on Kubernetes in an isolated environment. Every build gets its own container running on a dedicated Kubernetes node:
+Android workers run on Google Cloud Compute Engine machines in an isolated environment. Every build gets its own dedicated VM instance:
```
Comment (dsokal) [inline docs/pages/build-reference/infrastructure.mdx:32]: Don't mention Google Cloud. IMO this shouldn't be relevant to our users.

**Review by dsokal — APPROVED** [review]

File: docs/pages/build-reference/infrastructure.mdx (hunk 1 above)
Comment (szdziedzic) [reply docs/pages/build-reference/infrastructure.mdx:32]: I will just say:
`Android workers run on virtual machines in an isolated environment. Every build gets its own dedicated VM instance.`

Comment (szdziedzic) [reply docs/pages/build-reference/infrastructure.mdx:32]: What do you think?

File: docs/pages/build-reference/infrastructure.mdx [hunk 2]
```diff
@@ -151,9 +153,9 @@ iOS worker VMs run on Mac Mini 8.1 hosts in an isolated environment. Every build
 - Hardware:
   - Intel: Intel(R) Core(TM) i7-8700B CPU (6 cores/12 threads), 64 GB RAM
   - M1: 3.2GHz 8-Core M1, 16GB RAM [2020]
-- Build resource limits:
-  - Intel: 3 cores, 12 GB RAM
-  - M1: 3 cores, 8 GB RAM
+- Build resources:
+  - [`intel-medium`](eas-json/#resourceclass-2): 3 cores, 12 GB RAM
+  - [`m1-medium`](eas-json/#resourceclass-2): 3 cores, 8 GB RAM
```
Comment (wkozyra95) [inline docs/pages/build-reference/infrastructure.mdx:158]: is the number of cores here correct? Also what 3 cores in this context mean? M1 has performance and efficiency cores, so more detail can be useful in how resources are divided.

**Review by wkozyra95 — APPROVED** [review]

File: docs/pages/build-reference/infrastructure.mdx (hunk 2 above)
Comment (szdziedzic) [reply docs/pages/build-reference/infrastructure.mdx:158]: I will take a look at it

Comment (szdziedzic) [reply docs/pages/build-reference/infrastructure.mdx]: I found that our M1 hosts use the M1 processor with 8 cores (4 performance and 4 efficiency cores) and updated this information. I also noticed that actually, our M1 VMs use 2 cores vs 3 described above and updated it as well. 

Unfortunately, after quick research I wasn't able to find what using 2 cores per VM corresponds to in terms of access to performance and efficiency cores. Maybe @randomhajile has some ideas?

**Review by amandeepmittal — APPROVED** [review]
From doc's pov, lgtm! Thanks for making these changes.

File: docs/pages/build-reference/infrastructure.mdx [hunk 3]
```diff
@@ -146,14 +148,14 @@ Android workers run on Kubernetes in an isolated environment. Every build gets i
...
 - Hardware:
   - Intel: Intel(R) Core(TM) i7-8700B CPU (6 cores/12 threads), 64 GB RAM
-  - M1: 3.2GHz 8-Core M1, 16GB RAM [2020]
-- Build resource limits:
-  - Intel: 3 cores, 12 GB RAM
-  - M1: 3 cores, 8 GB RAM
+  - M1: 3.2GHz 8-Core M1 (4 performance and 4 efficiency cores), 16 GB RAM [2020]
+- Build resources:
+  - [`intel-medium`](eas-json/#resourceclass-2): 3 cores, 12 GB RAM
+  - [`m1-medium`](eas-json/#resourceclass-2): 2 cores, 8 GB RAM
```
Comment (szdziedzic) [inline docs/pages/build-reference/infrastructure.mdx:158]: @wkozyra95 What do you think about it? I added information about the fact that we run max 2 VMs per host.

File: docs/pages/build-reference/infrastructure.mdx (hunk 1 above)
Comment (dsokal) [reply docs/pages/build-reference/infrastructure.mdx]: Sounds good.

File: docs/pages/build-reference/infrastructure.mdx (hunk 3 above)
Comment (wkozyra95) [reply docs/pages/build-reference/infrastructure.mdx:158]: I would expect it in this line where the medium class is defined, but if we don't have a large class year it's probably fine for now.

**Pull request merged**